rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp

```diff
@@ -349,6 +349,19 @@ __declare_parameter_common(
     initial_value = &overrides_it->second;
   }
 
+  // If there is no initial value, then skip initialization
+  if (initial_value->get_type() == rclcpp::PARAMETER_NOT_SET) {
+    // descriptor type should be set
+    assert(parameter_descriptor.type != rclcpp::PARAMETER_NOT_SET);
```

Comment:
is this the case if `dynamic_typing == false`?

Reply:
Good catch! It is not always true, I missed this other instance where [`__declare_parameters_common()`](https://github.com/ros2/rclcpp/blob/8764151063988b4bec0680d857e648c17c2f9e47/rclcpp/src/rclcpp/node_interfaces/node_parameters.cpp#L656) is called. Compiling in debug mode causes a test failure :+1: 

The assertion is not necessary; removed in 24d2641